backend/game/scribble/handler/gamemaster/gamemaster_test.go

```diff
@@ -0,0 +1,160 @@
+package gamemaster
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/bseto/arcade/backend/game"
+	"github.com/bseto/arcade/backend/game/scribble/handler/gamemaster/util/point"
+	mockWf "github.com/bseto/arcade/backend/mocks/util/wordfactory"
+	mocks "github.com/bseto/arcade/backend/mocks/websocket/registry"
+	"github.com/bseto/arcade/backend/websocket/identifier"
+	"github.com/stretchr/testify/mock"
+)
+
+// TestWaitForStart will test that the gamemaster starts in the WaitForStart state.
+// Afterwards, it'll check to make sure that it will not proceed to the next
+// state unless the correct message has been passed through.
+func TestWaitForStart(t *testing.T) {
+	var reg mocks.Registry
+	var wordFactory mockWf.WordFactory
+	reg.On("SendToSameHubExceptCaller", mock.Anything, mock.Anything)
+	reg.On("SendToCaller", mock.Anything, mock.Anything)
+	wordFactory.On("GenerateWordList", 3).Return([]string{"a", "b", "c"})
+
+	gameMaster := Get(&reg)
+	ID := identifier.Client{
+		ClientUUID: identifier.ClientUUIDStruct{"AAA"},
+		HubName:    identifier.HubNameStruct{"BBB"},
+	}
+	gameMaster.NewClient(ID, &reg)
+	gameMaster.wordFactory = &wordFactory
+
+	if gameMaster.gameState != WaitForStart {
+		t.Errorf("got state: %v, expected: %v", gameMaster.gameState, WaitForStart)
+	}
+
+	receivedStruct := Receive{
+		GameMasterAPI: WaitForStart,
+		WaitForStartReceive: WaitForStartReceive{
+			StartGame: false,
+		},
+	}
+	b, err := json.Marshal(receivedStruct)
+	if err != nil {
+		t.Fatalf("unable to marshal struct: %v", err)
+	}
+	gameMaster.HandleInteraction(gameMaster.Name(), b, ID, &reg)
+	if gameMaster.gameState != WaitForStart {
+		t.Errorf("got state: %v, expected: %v", gameMaster.gameState, WaitForStart)
+	}
+
+	receivedStruct.WaitForStartReceive.StartGame = true
+	b, err = json.Marshal(receivedStruct)
+	if err != nil {
+		t.Fatalf("unable to marshal struct: %v", err)
+	}
+
+	gameMaster.HandleInteraction(gameMaster.Name(), b, ID, &reg)
+	time.Sleep(time.Millisecond * 100)
```

Comment:
not sure how to get around this since gameMaster has a `run()` loop that's constantly running. And when I call `gameMaster.HandleInteraction(gameMaster.Name(), b, ID, &reg)`, it delivers the message to the `run()` and then there it processes it and changes the state. So if I don't have a sleep, it'll just check the gamestate right away and fail

Reply:
> > Kudos, SonarCloud Quality Gate passed!
> > [<img alt="Bug" width="16" height="16" src="https://camo.githubusercontent.com/67545b394988d492007fcd881c0a94c5296edc29/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f6275672e706e67">](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=BUG) [<img alt="A" width="16" height="16" src="https://camo.githubusercontent.com/9701f681ab497016e6ae7f63fbaf44f4ab30d34c/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412e706e67">](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=BUG)
> > [<img alt="Vulnerability" width="16" height="16" src="https://camo.githubusercontent.com/475ba60a579e3b8ba1e2a251f4a264312f56d9b3/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f76756c6e65726162696c6974792e706e67">](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=VULNERABILITY) [<img alt="A" width="16" height="16" src="https://camo.githubusercontent.com/9701f681ab497016e6ae7f63fbaf44f4ab30d34c/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412e706e67">](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=VULNERABILITY) (and [<img alt="Security Hotspot" width="16" height="16" src="https://camo.githubusercontent.com/cdf79f6d9d06866802c83878aa3be0bd576eb7f9/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f73656375726974795f686f7473706f742e706e67">](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=SECURITY_HOTSPOT) to review)
> > [<img alt="Code Smell" width="16" height="16" src="https://camo.githubusercontent.com/8747c0c3ceeb057fd2942c9cef31347d3bcbb6d4/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f636f64655f736d656c6c2e706e67">](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=CODE_SMELL) [<img alt="A" width="16" height="16" src="https://camo.githubusercontent.com/9701f681ab497016e6ae7f63fbaf44f4ab30d34c/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412e706e67">](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=CODE_SMELL) [14 Code Smells](https://sonarcloud.io/project/issues?id=nightowl-studios_arcade&pullRequest=233&resolved=false&types=CODE_SMELL)
> > [<img alt="6.7%" width="16" height="16" src="https://camo.githubusercontent.com/6729652312bc75aad1adaa53ca441c7538a01edc/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f436f76657261676543686172742f302e706e67">](https://sonarcloud.io/component_measures?id=nightowl-studios_arcade&pullRequest=233&metric=new_coverage&view=list) [6.7% Coverage](https://sonarcloud.io/component_measures?id=nightowl-studios_arcade&pullRequest=233&metric=new_coverage&view=list)
> > [<img alt="0.0%" width="16" height="16" src="https://camo.githubusercontent.com/540735ca9aeb74d34ef0758d1adaeeb0d1c173e7/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f4475706c69636174696f6e732f332e706e67">](https://sonarcloud.io/component_measures?id=nightowl-studios_arcade&pullRequest=233&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=nightowl-studios_arcade&pullRequest=233&metric=new_duplicated_lines_density&view=list)
> 
> All of the code smells are from the Mocks (I checked a lot of them, but didn't click through them all)

You can add a regex for the mock folder to `sonar.exclusions` (probably something like `**/mocks/*.go`

Reply:
Probably fine for now